### PR TITLE
Add all mutations

### PR DESCRIPTION
**Pull request by tomermoshe**

### CODE REVIEW

File: packages/graphql-client/src/graphql-client.js [hunk 1]
```diff
@@ -89,16 +89,16 @@ export class GraphQLClient {
     return await this.mutate(verifyEmailMutation, 'verifyEmail', { token });
   }
 
-  async resetPassword(token: string, newPassword: string): Promise<void> {
+  async resetPassword(token: string, newPassword: any): Promise<void> {
```
Comment (davidyaha) [inline packages/graphql-client/src/graphql-client.js]: Why `any`?

File: packages/graphql-client/src/graphql-client.js [hunk 2]
```diff
@@ -89,16 +89,16 @@ export class GraphQLClient {
...
 
-  async sendVerificationEmail(userId: string, email: string): Promise<void> {
-    return await this.mutate(sendVerificationEmailMutation, 'sendVerificationEmail', { userId, email });
+  async sendVerificationEmail(email: string): Promise<void> {
+    return await this.mutate(sendVerificationEmailMutation, 'sendVerificationEmail', { email });
   }
 
-  async sendResetPasswordEmail(userId: string, email: string): Promise<void> {
-    return await this.mutate(sendResetPasswordEmailMutation, 'sendResetPasswordEmail', { userId, email });
+  async sendResetPasswordEmail(email: string): Promise<void> {
```
Comment (davidyaha) [inline packages/graphql-client/src/graphql-client.js]: I think we should change the email field to email or username or userId. It will allow different implementations here (admin console, batch reset and so on). It should be same input as the login I think.

File: packages/graphql-client/src/graphql-client.js [hunk 3]
```diff
@@ -89,16 +89,16 @@ export class GraphQLClient {
...
   }
 
-  async resetPassword(token: string, newPassword: string): Promise<void> {
+  async resetPassword(token: string, newPassword: any): Promise<void> {
     return await this.mutate(resetPasswordMutation, 'resetPassword', { token, newPassword });
   }
 
-  async sendVerificationEmail(userId: string, email: string): Promise<void> {
-    return await this.mutate(sendVerificationEmailMutation, 'sendVerificationEmail', { userId, email });
+  async sendVerificationEmail(email: string): Promise<void> {
```
Comment (davidyaha) [inline packages/graphql-client/src/graphql-client.js]: The only use cases I can think of for the client to request verification email, are if you are enforcing login for verified only users, if you have non-verified status and the user wants to verify, or lastly, if an admin will run this request for user. In the first cases, the user is technically logged-in in some way, and in the later case the admin can identify users with whatever is available on the client (userId, email or username). This means that the input should probably be, optional complex type that can have either of those.

File: packages/graphql-api/src/graphql/types.js [hunk 4]
```diff
@@ -21,4 +21,26 @@ export const typeDefs = `
...
+  input UserInput {
+    id: ID
+    email: String
+    username: String
+  }
+  
+  input CreateUserInput {
+    username: String
+    email: String
+    password: PasswordType
```
Comment (davidyaha) [inline packages/graphql-api/src/graphql/types.js]: You cannot use types within input types.

File: packages/graphql-api/src/graphql/types.js [hunk 5]
```diff
@@ -21,4 +21,26 @@ export const typeDefs = `
     email: String
     username: String
   }
+  
+  input UserInput {
```
Comment (davidyaha) [inline packages/graphql-api/src/graphql/types.js:25]: Maybe create input type for each mutation.

**Review by davidyaha — CHANGES_REQUESTED** [review]

File: packages/graphql-api/src/graphql/types.js (hunk 5 above)
Comment (tomermoshe) [reply packages/graphql-api/src/graphql/types.js:25]: why?

File: packages/graphql-client/src/graphql-client.js (hunk 3 above)
Comment (tomermoshe) [inline packages/graphql-client/src/graphql-client.js]: Lets do it in a different PR as it also requires changes in the server

File: packages/graphql-api/src/graphql/types.js (hunk 5 above)
Comment (davidyaha) [reply packages/graphql-api/src/graphql/types.js:25]: https://dev-blog.apollodata.com/designing-graphql-mutations-e09de826ed97

File: packages/graphql-client/src/graphql-client.js (hunk 3 above)
Comment (davidyaha) [inline packages/graphql-client/src/graphql-client.js]: 👍

**Review by davidyaha — APPROVED** [review]
